torch/_dynamo/eval_frame.py

```diff
@@ -1608,7 +1603,14 @@ def graph_with_interpreter(*args):
 
         if same_signature:
             flat_args_dynamic_dims = [
-                {c.dim for c in (constraints or ()) if c.w_tensor() is x}
```

Comment:
needed to keep dims on unused shapes as static

Reply:
you'll probably hit a merge conflict here, I changed this to use `c.t_id` instead